src/test/storage/dictionary_segment_test.cpp

```diff
@@ -70,6 +70,127 @@ TEST_F(StorageDictionarySegmentTest, LowerUpperBound) {
+  value_segment_int->append(1);
+  value_segment_int->append(2);
+
+  std::shared_ptr<AbstractSegment> segment;
+  resolve_data_type("int", [&](auto type) {
+    using Type = typename decltype(type)::type;
+    segment = std::make_shared<DictionarySegment<Type>>(value_segment_int);
+  });
+  auto dict_segment = std::dynamic_pointer_cast<DictionarySegment<int32_t>>(segment);
+
+  EXPECT_ANY_THROW(dict_segment->append(AllTypeVariant{5}));
+}
+
+TEST_F(StorageDictionarySegmentTest, AccessingUnderlyingDataStructures) {
+  value_segment_int->append(1);
+  value_segment_int->append(2);
+  value_segment_int->append(2);
+
+  std::shared_ptr<AbstractSegment> segment_int;
+  resolve_data_type("int", [&](auto type) {
+    using Type = typename decltype(type)::type;
+    segment_int = std::make_shared<DictionarySegment<Type>>(value_segment_int);
+  });
+  auto dict_segment_int = std::dynamic_pointer_cast<DictionarySegment<int32_t>>(segment_int);
+
+  std::shared_ptr<AbstractSegment> segment_str;
+  resolve_data_type("string", [&](auto type) {
+    using Type = typename decltype(type)::type;
+    segment_str = std::make_shared<DictionarySegment<Type>>(value_segment_str);
+  });
+  auto dict_segment_str = std::dynamic_pointer_cast<DictionarySegment<std::string>>(segment_str);
+
+  // test accessing the underlying dictionary
+  EXPECT_EQ(dict_segment_int->dictionary(), (std::vector<int>{1, 2}));
+  EXPECT_EQ(dict_segment_str->dictionary(), (std::vector<std::string>{}));
+
+  // test accessing the underlying attribute vector
+  auto att_vec_int = dict_segment_int->attribute_vector();
+  auto att_vec_str = dict_segment_str->attribute_vector();
+  EXPECT_EQ(att_vec_int->get(0), int32_t{0});
+  EXPECT_EQ(att_vec_int->get(1), int32_t{1});
+  EXPECT_EQ(att_vec_int->get(2), int32_t{1});
+  EXPECT_ANY_THROW(att_vec_str->get(0));
+}
+
+TEST_F(StorageDictionarySegmentTest, ValueOfValueId) {
+  value_segment_int->append(1);
+  value_segment_int->append(2);
+  value_segment_int->append(2);
+
+  std::shared_ptr<AbstractSegment> segment;
+  resolve_data_type("int", [&](auto type) {
+    using Type = typename decltype(type)::type;
+    segment = std::make_shared<DictionarySegment<Type>>(value_segment_int);
+  });
+  auto dict_segment = std::dynamic_pointer_cast<DictionarySegment<int32_t>>(segment);
+
+  EXPECT_EQ(dict_segment->value_of_value_id(ValueID{0}), 1);
+  EXPECT_EQ(dict_segment->value_of_value_id(ValueID{1}), 2);
+  EXPECT_ANY_THROW(dict_segment->value_of_value_id(ValueID{2}));
+}
+
+TEST_F(StorageDictionarySegmentTest, UInt8) {
+  for (int32_t index = 0; index < 100; ++index) {
+    value_segment_int->append(index);
+  }
+  value_segment_int->append(42);
+  auto column = std::make_shared<DictionarySegment<int32_t>>(value_segment_int);
+  auto dict_col = std::dynamic_pointer_cast<opossum::DictionarySegment<int32_t>>(column);
+
+  EXPECT_EQ(dict_col->estimate_memory_usage(), 100 * sizeof(int32_t) + 101 * sizeof(uint8_t));
```

Comment:
It would be better to explicitly test the ``estimate_memory_usage`` function, just to ensure that it does not contain errors which then taint the test results here.